zap/src/main/java/org/zaproxy/zap/extension/brk/BreakPanelToolbarFactory.java

```diff
@@ -56,18 +62,34 @@
     private BreakResponsesButtonAction breakResponsesButtonAction;
     private BreakAllButtonAction breakAllButtonAction;
 
+    private SetBreakOnJavaScriptAction setBreakOnJavaScriptAction;
+    private SetBreakOnCSSAndFontsAction setBreakOnCSSAndFontsAction;
+    private SetBreakOnMultimediaAction setBreakOnMultimediaAction;
+    private SetBreakOnlyOnScopeAction setOnlyBreakOnScopeAction;
+
     private boolean cont = false;
     private boolean step = false;
     private boolean stepping = false;
     private boolean drop = false;
     private boolean isBreakRequest = false;
     private boolean isBreakResponse = false;
     private boolean isBreakAll = false;
+    private boolean isBreakOnJavaScript = true;
+    private boolean isBreakOnCSSAndFonts = true;
+    private boolean isBreakOnMultimedia = true;
+    private boolean isOnlyBreakOnScope = false;
```

Comment:
This should be replaced with the use of the existing option (`BreakpointsParam#isInScopeOnly()`), otherwise we might end up with conflicting/misleading state, one set to break and the other unset.

Reply:
ok. Should I also add isBreakOnJavaScript, isBreakOnCSSAndFonts and isBreakOnMultimedia to the  breakpointsParams, or I should only remove the isOnlyBreakOnScope?

Reply:
I don't think they need to be persisted, so fine leaving them here.

> maybe the inScopeOnly param should be removed from the BreakpointsParam instead.

There are other scope related options (e.g. passive scan just in scope) that are persisted, it would be better to preserve the current behaviour.